src/main/java/org/heigit/ohsome/ohsomeapi/executor/ExecutionUtils.java

```diff
@@ -391,28 +392,17 @@ public void writeCsvResponse(Result[] resultSet, HttpServletResponse servletResp
 
   /** Creates the <code>Feature</code> objects in the OSM data response. */
   public org.wololo.geojson.Feature createOSMFeature(OSMEntity entity, Geometry geometry,
-      Map<String, Object> properties, int[] keysInt, boolean includeTags,
+      Map<String, Object> properties, Set<Integer> keysInt, boolean includeTags,
       boolean includeOSMMetadata, ElementsGeometry elemGeom) {
     if (geometry.isEmpty()) {
       // skip invalid geometries (e.g. ways with 0 nodes)
       return null;
     }
     if (includeTags) {
-      for (OSHDBTag oshdbTag : entity.getTags()) {
-        properties.put(String.valueOf(oshdbTag.getKey()), oshdbTag);
-      }
-    } else if (keysInt.length != 0) {
-      int[] tags = entity.getRawTags();
-      for (int i = 0; i < tags.length; i += 2) {
-        int tagKeyId = tags[i];
-        int tagValueId = tags[i + 1];
-        for (int key : keysInt) {
-          if (tagKeyId == key) {
-            properties.put(String.valueOf(tagKeyId), new OSHDBTag(tagKeyId, tagValueId));
-            break;
-          }
-        }
-      }
+      properties.put("@tags", Iterables.toArray(entity.getTags(), OSHDBTag.class));
+    } else if (!keysInt.isEmpty()) {
+      Iterable<OSHDBTag> filtered = Iterables.filter(entity.getTags(), t -> keysInt.contains(t.getKey()));
```

Comment:
This line here is longer than the maximal allowed line length defined by our checkstyle file.